src/System.Transactions/tests/EnlistTest.cs

```diff
@@ -0,0 +1,1193 @@
+        }
+
+        [Fact]
+        public void Vol2_Dur1_Fail5()
+        {
+            CommittableTransaction ct = new CommittableTransaction();
+            IntResourceManager[] irm = new IntResourceManager[2];
+            irm[0] = new IntResourceManager(1);
+            irm[1] = new IntResourceManager(3);
+
+            Transaction.Current = ct;
+            irm[0].Type = ResourceManagerType.Durable;
+            irm[0].FailSPC = true;
+            irm[0].FailWithException = true;
+
+            for (int i = 0; i < 2; i++)
+                irm[i].UseSingle = true;
+
+            /* Durable RM irm[2] does on SPC, so
+			 * all volatile RMs get Rollback */
+
+            using (TransactionScope scope = new TransactionScope())
+            {
+                irm[0].Value = 2;
+                irm[1].Value = 6;
+
+                scope.Complete();
+            }
+
+            TransactionAbortedException tae = Assert.Throws<TransactionAbortedException>(() => ct.Commit());
+            Assert.IsType<NotSupportedException>(tae.InnerException);
+
+            irm[0].Check(1, 0, 0, 0, 0, 0, 0, "irm [0]");
+            irm[1].Check(0, 1, 0, 1, 0, 0, 0, "irm [1]");
+
+            InvalidOperationException ioe = Assert.Throws<InvalidOperationException>(() => ct.Commit());
+            Assert.Null(ioe.InnerException);
+
+            Transaction.Current = null;
+        }
+
+        #endregion
+
+        #region Promotable Single Phase Enlistment
+        [Fact]
+        public void Vol0_Dur0_Pspe1()
+        {
+            IntResourceManager irm = new IntResourceManager(1);
+            irm.Type = ResourceManagerType.Promotable;
+            using (TransactionScope scope = new TransactionScope())
+            {
+                irm.Value = 2;
+
+                scope.Complete();
+            }
+            irm.Check(1, 0, 0, 0, 0, 1, 0, "irm");
+        }
+
+        [Fact]
+        public void Vol1_Dur0_Pspe1()
+        {
+            IntResourceManager irm0 = new IntResourceManager(1);
+            IntResourceManager irm1 = new IntResourceManager(1);
+            irm1.Type = ResourceManagerType.Promotable;
+            using (TransactionScope scope = new TransactionScope())
+            {
+                irm0.Value = 2;
+                irm1.Value = 8;
+
+                scope.Complete();
+            }
+            irm1.Check(1, 0, 0, 0, 0, 1, 0, "irm1");
+        }
+
+        [Fact]
+        public void Vol0_Dur1_Pspe1()
+        {
+            IntResourceManager irm0 = new IntResourceManager(1);
+            IntResourceManager irm1 = new IntResourceManager(1);
+            irm0.Type = ResourceManagerType.Durable;
+            irm0.UseSingle = true;
+            irm1.Type = ResourceManagerType.Promotable;
+            using (TransactionScope scope = new TransactionScope())
+            {
+                irm0.Value = 8;
+                irm1.Value = 2;
+                Assert.Equal(0, irm1.NumEnlistFailed);
```

Comment:
This was all just ported as-is from the Mono tests, albeit with xunit instead of another harness' assert methods being used.  I'll add a TODO with your comment.